### PR TITLE
create a specific implementation for Idle Offline Cause

### PR DESCRIPTION
**Pull request by varmenise**
@reviewbybees

### CODE REVIEW

**Comment by rsandell** [comment]
:bee:

File: core/src/main/java/hudson/slaves/OfflineCause.java [hunk 1]
```diff
@@ -155,4 +155,14 @@ public ByCLI(String message) {
             this.message = message;
         }
     }
+
+    /**
+     * Caused by idle period.
```
Comment (KostyaSha) [inline core/src/main/java/hudson/slaves/OfflineCause.java]: Description should be descriptive imho. `Offline caused by idle period` or `Offline caused by idle timeout`

**Comment by KostyaSha** [comment]
:+1: for unbundling anonymous classes

**Comment by oleg-nenashev** [comment]
I agree with @KostyaSha 's comment regarding Javadoc, but :bee: in any case

**Comment by varmenise** [comment]
@oleg-nenashev @KostyaSha I agree with you but I had preferred to do that way since all the other `OfflineCause` classes have that kind of description (e.g `Caused by unexpected channel termination.` and all the others)

**Comment by KostyaSha** [comment]
@reviewbybees done ?

**Pull request merged**